### DISCUSSION TITLE
Expanding arrays to be used as exec arguments

### DISCUSSION THREAD

**Discussion by aoinokimi** [sagiegurari/duckscript#464] [Q&A]
I have the following script that I run as a cargo make task:

```
[tasks.myprogram]
script_runner = "@duckscript"
script = ['''
arguments = set ${@}
echo "Running myprogram ${arguments}"
exec --fail-on-error myprogram ${arguments}
''']
```

In the exec line, when I execute my task, ${arguments} gets resolved as a handle (something like handle:3mBwDvRbEa4H4BKmoKMv) instead of the actual elememts of my array. How can I make this array get expanded? I tried creating an string joining all elements of array leaving an space between elements, but then the whol string is considrered as a single parameter by exec. Any suggestion on how to handle this will be much appreciated.

**Comment by sagiegurari** [comment]
you could use array_join with " " to make it a line and than in exec use the % binding to split it to words

**Reply by aoinokimi** [reply]
Thank you for the tip. This mostly works. Still, this doesn't cover all my needs. If I do something like: cargo make myprogram -arg1=a -arg2="a b c", my "a b c" string gets splited into three separate tokens when I use %, breaking the incoation to myprgram. Is there any way to easily address this?
